wgengine/router/dns/nm.go

```diff
@@ -52,15 +53,82 @@ func isNMActive() bool {
 
 // nmManager uses the NetworkManager DBus API.
 type nmManager struct {
+	logf          logger.Logf
 	interfaceName string
 }
 
 func newNMManager(mconfig ManagerConfig) managerImpl {
 	return nmManager{
+		logf:          mconfig.Logf,
 		interfaceName: mconfig.InterfaceName,
 	}
 }
 
+const nmDeviceActiveState = 100
+
+// awaitDeviceActive returns when the NetworkManager device at devicePath becomes active
+// or when ctx expires, whichever happens first.
+func (m nmManager) awaitDeviceActive(ctx context.Context, conn *dbus.Conn, devicePath dbus.ObjectPath) error {
```

Comment:
How long can this block? I'm nervous about adding blocking states in the reconfig path, because it's really easy for us to accidentally trip the wgengine watchdog and crash.